dace/transformation/dataflow/buffer_tiling.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
+""" Contains classes that implement the OnTheFlyMapFusion transformation. """
+
+from dace import registry
+from dace.sdfg import nodes
+from dace.sdfg import utils as sdutil
+from dace.transformation import transformation
+from dace.transformation.dataflow import MapTiling, MapTilingWithOverlap, MapFusion, TrivialMapElimination
+
+@registry.autoregister_params(singlestate=True)
+class BufferTiling(transformation.Transformation):
+    _map1_exit = nodes.MapExit(nodes.Map('', [], []))
+    _array = nodes.AccessNode('')
+    _map2_entry = nodes.MapEntry(nodes.Map('', [], []))
+
+    # Returns a list of graphs that represent the pattern
+    @staticmethod
+    def expressions():
+        return [
+            sdutil.node_path_graph(
+                BufferTiling._map1_exit,
+                BufferTiling._array,
+                BufferTiling._map2_entry,
+            )
+        ]
+
+    @staticmethod
+    def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
+        map1_exit = graph.nodes()[candidate[BufferTiling._map1_exit]]
+        map2_entry = graph.nodes()[candidate[BufferTiling._map2_entry]]
+
+        # Check that locally provided buffer has volume 1.
+        for edge in graph.in_edges(map1_exit):
+            if edge.data.volume != 1:
+                return False
+
+        # Check that all locally provided buffers have the same subset.
+        subsets = set(edge.data.subset for edge in graph.in_edges(map1_exit))
+        if len(subsets) != 1:
+            return False
+
+        # Check existence of buffers
+        buffers = graph.all_nodes_between(map1_exit, map2_entry)
+        if buffers is None:
+            return False
+
+        for buf in buffers:
+            # Check that buffers are AccessNodes.
+            if not isinstance(buf, nodes.AccessNode):
+                return False
+
+            # Check that buffers are transient.
+            if not sdfg.arrays[buf.data].transient:
+                return False
+            
+            # Check that buffers have exactly 1 input and 1 output edge.
+            if graph.in_degree(buf) != 1:
+                return False
+            if graph.out_degree(buf) != 1:
+                return False
+            
+            # Check that buffers are next to the maps.
+            if graph.in_edges(buf)[0].src != map1_exit:
+                return False
+            if graph.out_edges(buf)[0].dst != map2_entry:
+                return False
+
+            # Check that the data consumed is provided.
+            provided = graph.in_edges(buf)[0].data.subset
+            consumed = graph.out_edges(buf)[0].data.subset
+            if not provided.covers(consumed):
+                return False
+
+            # Check that buffers occure only once in this state.
+            num_occurrences = len([
```

Comment:
why not check if it occurs only once in the entire SDFG?

Reply:
To be conform with [MapFusion](https://github.com/spcl/dace/blob/master/dace/transformation/dataflow/map_fusion.py#L112).

Reply:
But maybe we should change both. But I'm not 100% sure though.